core-bundle/src/Twig/Interop/ContaoEscaper.php

```diff
@@ -43,10 +60,85 @@ public function __invoke(Environment $environment, $string, ?string $charset): s
             $string
         );
 
+        return htmlspecialchars($string, ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8', false);
+    }
+
+    /**
+     * This implementation is a clone of Twig's html_attr escape strategy but
+     * replaces insert tags and decodes entities beforehand.
+     *
+     * @see twig_escape_filter
+     */
+    public function escapeHtmlAttr(Environment $environment, $string, ?string $charset): string
+    {
         if (null !== $charset && 'UTF-8' !== strtoupper($charset)) {
-            throw new RuntimeError(sprintf('The "contao_html" escape filter does not support the %s charset, use UTF-8 instead.', $charset));
+            throw new RuntimeError(sprintf('The "contao_html_attr" escape filter does not support the %s charset, use UTF-8 instead.', $charset));
         }
 
-        return htmlspecialchars($string, ENT_QUOTES | ENT_SUBSTITUTE, 'UTF-8', false);
+        $string = (string) $string;
+
+        // Replace insert tags before '{' and '}' get encoded
+        $string = $this->framework->getAdapter(Controller::class)->replaceInsertTags($string);
```

Comment:
```suggestion
        $string = $this->framework->getAdapter(Controller::class)->replaceInsertTags($string, false);
```

`$blnCache` needs to be `false` I think, otherwise uncachable insert tags will not get replaced.